Implementation/Services/index.md

```diff
@@ -0,0 +1,510 @@
+:::
+
+## Accessing Core Services and Helpers when there is no UmbracoContext eg in a Component or C# Class
+
+Controllers and Views have an UmbracoContext, however this is not always the case 'everywhere in Umbraco', for example common extension points: Components,ContentFinders or Custom C# Classes.
+
+:::Warning
+If the UmbracoContext is not guaranteed to exist you cannot inject any helpers or services that rely on the UmbracoContext, eg UmbracoHelper, PublishedContentQuery and if you try to inject these helpers into the constructors of these context less classes...  Umbraco will report a 'boot' error on startup.
+:::
+
+### Injecting Services into a Component
+
+It's possible to inject Management Services that do not rely on the `UmbracoContext` into the constructor of a component. This example shows injecting the MediaService in a Component to create a corresponding Media Folder for every 'landing page' that is saved in the Content Section, by subscribing to the 'Content Saved' event. 
+
+```csharp
+using System;
+using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+
+namespace Umbraco8.Components
+{
+    [RuntimeLevel(MinLevel = RuntimeLevel.Run)]
+    public class SubscribeToContentSavedEventComposer : ComponentComposer<SubscribeToContentSavedEventComponent>
+    {
+    }
+
+    public class SubscribeToContentSavedEventComponent : IComponent
+    {
+        private readonly IMediaService _mediaService;
+        public SubscribeToContentSavedEventComponent(IMediaService mediaService)
+        {
+            _mediaService = mediaService;
+        }
+        public void Initialize()
+        {
+            ContentService.Saved += ContentService_Saved;
+        }
+
+        private void ContentService_Saved(Umbraco.Core.Services.IContentService sender, Umbraco.Core.Events.ContentSavedEventArgs e)
+        {
+            foreach (var contentItem in e.SavedEntities)
+            {
+                //if this is a new landing page create a folder for associated media in the media section
+                if (contentItem.ContentType.Alias == "landingPage")
+                {
+                    // we have injected in the mediaService in the contstructor for the component see above.
+                   bool hasExistingFolder = _mediaService.GetByLevel(1).Any(f => f.Name == contentItem.Name);
+                   if (!hasExistingFolder)
+                    {
+                        //let's create one (-1 indicates the root of the media section)
+                       IMedia newFolder = _mediaService.CreateMedia(contentItem.Name, -1, "Folder");
+                        _mediaService.Save(newFolder);
+                    }
+                }
+            }
+        }       
+
+        public void Terminate()
+        {
+            throw new NotImplementedException();
```

Comment:
```suggestion
```

Recommend you remove this @marcemarc as when Umbraco tries to gracefully shutdown its gonna YSOD